rosidl_generator_py/resource/_msg.py.em

```diff
@@ -338,7 +338,12 @@ if isinstance(type_, AbstractNestedType):
         for s, t in zip(self.__slots__, self.SLOT_TYPES):
             field = getattr(self, s)
             fieldstr = repr(field)
-            if isinstance(t, rosidl_parser.definition.AbstractSequence):
+            # We use Python array type for fields that can be directly stored
+            # in them, and "normal" sequences for everything else.  If it is
+            # a type that we store in an array, strip off the 'array' portion.
+            if isinstance(t, rosidl_parser.definition.AbstractSequence) and \
+               isinstance(t.value_type, rosidl_parser.definition.BasicType) and \
+               t.value_type.typename in @([*SPECIAL_NESTED_BASIC_TYPES]):
```

Comment:
Nitpick: the multi line condition isn't distinguishable from the indented block. Please use the following style instead:

```
if (
    ... and
    ...
):
    ...
```

Reply:
Done in f7b18f6e065cea676e8a13d1a3a83043094f307a